querybook/server/lib/query_analysis/templating.py

```diff
@@ -26,179 +30,241 @@ class QueryJinjaSyntaxException(QueryTemplatingError):
     pass
 
 
+class LatestPartitionFunctionException(QueryTemplatingError):
+    pass
+
+
 # The first part is regex for single line comment, ie -- some comment
 # the second part is for multi line comment, ie /* test */
 comment_re = re.compile(r"((?:--.*)|(?:\/\*(?:.|\n)*?\*\/))", re.MULTILINE)
 
 
-def _escape_sql_comments(query: str):
-    return re.sub(
-        comment_re, lambda match: "{{ " + json.dumps(match.group()) + " }}", query
-    )
+class TemplatedQueryRenderer(object):
+    def __init__(self, engine_id: int = None):
+        self.env = SandboxedEnvironment()
+        self.engine_id = engine_id
+        self.env.globals.update(latest_partition=self.get_latest_partition)
```

Comment:
to share env, i guess we can construct an env in render_templated_query and share it across different calls